roles/contrail_common/tasks/determine_contrail_masters.yaml

```diff
@@ -0,0 +1,25 @@
+---
+- name: Set contrail_masters var
+  set_fact:
+    contrail_masters: ""
+
+- name: Set contrail_masters from contrail_control_data_interface
+  set_fact:
+    contrail_masters: "{{ contrail_control_data_interface }}"
+  when:
+    - contrail_control_data_interface is defined
+
+- name: Set contrail_masters from infra nodes
+  set_fact:
+    contrail_masters: "{{ contrail_masters }} {{ item }}"
+  when:
+    - contrail_control_data_interface is not defined
+    - hostvars[item]['openshift_node_group_name'] is defined
+    - hostvars[item]['openshift_node_group_name'] == 'node-config-infra'
```

Comment:
Users may use custom group names, it should be a var

Reply:
for now we support only installing on infra node, so we would like to have this as is, Thanks

Reply:
There is a default node group for infra nodes, yes, but any node with `infra` label would run infra containers - e.g. a host with `node-config-all-in-one` group or any user-defined group.

Reply:
we have not tested all-in-one or installing on user groups. For now we install on "node-config-infra" label only. It requires more engineering and testing to support, so can we have this in next commit as our customer is already asking for this ?